### DISCUSSION TITLE
GraphQL Depth Limit MaxQueryDepthInstrumentation is there a way to expose this class as directive or annotation?

### DISCUSSION THREAD

**Discussion by sbilello** [Netflix/dgs-framework#245] [Q&A] [answered]
https://github.com/graphql-java/graphql-java/blob/master/src/main/java/graphql/analysis/MaxQueryDepthInstrumentation.java
I was looking at this class inside the graphql-java core: Is there a corresponding directive that can be added to the schema?

I find many projects in javascript to deal with query cost analysis, amount limit, depth limit 
What's the corresponding approach in the Java world if there is any?
I would like to cover all the aspects mentioned over here [graphql-api-security](https://leapgraph.com/graphql-api-security) but using java instead of javascript

**Comment by srinivasankavitha** [comment, marked as answer]
You can add an instrumentation class that extends the MaxQueryDepthInstrumentation class: https://netflix.github.io/dgs/advanced/instrumentation/

This will allow you to limit the query depth.

Hope that helps.
